crates/redstone/benches/benchmarks.rs

```diff
@@ -0,0 +1,82 @@
+    "ZAB", "ACB", "AAA", "ABA", "ADC", "DCA", "XYZ", "XYZD", "ABC", "ACC", "ABB", "ABCD", "CDE",
+    "BCD", "0RT", "ZAB", "ACB", "AAA", "ABA", "ADC", "DCA", "XYZ", "XYZD", "ABC", "ACC", "ABB",
+    "ABCD", "CDE", "BCD", "0RT", "ZAB", "ACB", "AAA", "ABA", "ADC", "DCA", "XYZ", "XYZD", "ABC",
+    "ACC", "ABB", "ABCD", "CDE", "BCD", "0RT", "ZAB", "ACB", "AAA", "ABA", "ADC", "DCA", "XYZ",
+    "XYZD", "ABC", "ACC", "ABB", "ABCD", "CDE", "BCD", "0RT", "ZAB", "ACB", "AAA", "ABA", "ADC",
+    "DCA", "XYZ", "XYZD", "ABC", "ACC", "ABB", "ABCD", "CDE", "BCD", "0RT", "ZAB", "ACB", "AAA",
+    "ABA", "ADC", "DCA", "XYZ", "XYZD", "ABC", "ACC", "ABB", "ABCD", "CDE", "BCD", "0RT", "ZAB",
+    "ACB", "AAA", "ABA", "ADC", "DCA", "XYZ", "XYZD", "ABC", "ACC", "ABB", "ABCD", "CDE", "BCD",
+    "0RT", "ZAB", "ACB", "AAA", "ABA", "ADC", "DCA", "XYZ", "XYZD", "ABC", "ACC", "ABB", "ABCD",
+    "CDE", "BCD", "0RT", "ZAB", "ACB", "AAA", "ABA", "ADC", "DCA", "XYZ", "XYZD", "ABC", "ACC",
+    "ABB", "ABCD", "CDE", "BCD", "0RT", "ZAB", "ACB", "AAA", "ABA", "ADC", "DCA", "XYZ", "XYZD",
+    "ABC", "ACC", "ABB", "ABCD", "CDE", "BCD", "0RT", "ZAB", "ACB", "AAA", "ABA", "ADC", "DCA",
+    "XYZ", "XYZD", "ABC", "ACC", "ABB", "ABCD", "CDE", "BCD", "0RT", "ZAB", "ACB", "AAA", "ABA",
+    "ADC", "DCA", "XYZ", "XYZD", "ABC", "ACC", "ABB", "ABCD", "CDE", "BCD", "0RT", "ZAB", "ACB",
+    "AAA", "ABA", "ADC", "DCA", "XYZ", "XYZD", "ABC", "ACC", "ABB", "ABCD", "CDE", "BCD", "0RT",
+    "ZAB", "ACB", "AAA", "ABA", "ADC", "DCA", "XYZ", "XYZD", "ABC", "ACC", "ABB", "ABCD", "CDE",
+    "BCD", "0RT", "ZAB", "ACB", "AAA", "ABA", "ADC", "DCA", "XYZ", "XYZD", "ABC", "ACC", "ABB",
+    "ABCD", "CDE", "BCD", "0RT", "ZAB", "ACB", "AAA", "ABA", "ADC", "DCA", "XYZ", "XYZD", "ABC",
+    "ACC", "ABB", "ABCD", "CDE", "BCD", "0RT", "ZAB", "ACB", "AAA", "ABA", "ADC", "DCA", "XYZ",
+    "XYZD", "ABC", "ACC", "ABB", "ABCD", "CDE", "BCD", "0RT", "ZAB", "ACB", "AAA", "ABA", "ADC",
+    "DCA", "XYZ", "XYZD", "ABC", "ACC", "ABB", "ABCD", "CDE", "BCD", "0RT", "ZAB", "ACB", "AAA",
+    "ABA", "ADC", "DCA", "XYZ", "XYZD", "ABC", "ACC", "ABB", "ABCD", "CDE", "BCD", "0RT", "ZAB",
+    "ACB", "AAA", "ABA", "ADC", "DCA", "XYZ", "XYZD",
+];
+
+fn benchmark_trie(c: &mut Criterion) {
+    c.bench_function("benchmark_trie", |b| {
+        let feeds: Vec<FeedId> = TEST_FEED_IDS
+            .iter()
+            .map(|f| f.as_bytes().to_vec().into())
+            .collect();
+        b.iter(|| {
+            let mut trie = Trie::default();
+            for feed in feeds.iter() {
+                let _ = trie.store(feed);
+            }
+        });
+    });
+}
+
+fn benchmark_hashset(c: &mut Criterion) {
+    c.bench_function("benchmark_hashset", |b| {
+        let feeds: Vec<FeedId> = TEST_FEED_IDS
+            .iter()
+            .map(|f| f.as_bytes().to_vec().into())
+            .collect();
+        b.iter(|| {
+            let mut hs = HashSet::new();
```

Comment:
HashSet with default `RandomState` is slow. I have tried the FxHashSet from https://github.com/rust-lang/rustc-hash and in this benchmark its twice as fast compared to Trie, but the hashset is available onkly on std.

Reply:
It might be the case of the Trie implementation - not optimized enough. The main reason for not using the `hashset` is its larger memory footprint for larger datasets.  But I assume that we will never have more `data_points` than `0xFFFF` as two bytes encodes the `count`.
That being said, I am totally ok with removing the Trie and stick to something simple and well tested such as `hashset`. 
The argument for the `hashset` is that we don't have to worry about the encoding (`ASCII` vs `UTF8` vs `UTF16`) if there will be a Chinese character in feed_id for whatever reason it will still work just fine.